kymatio/scattering1d/frontend/torch_frontend.py

```diff
@@ -139,8 +143,130 @@ def scattering(self, x):
 
         return S
 
-
 ScatteringTorch1D._document()
 
 
-__all__ = ['ScatteringTorch1D']
+class TimeFrequencyScatteringTorch1D(TimeFrequencyScatteringBase1D,
+                                     ScatteringTorch1D):
+    def __init__(self, J, shape, Q, J_fr=None, Q_fr=2, T=None, F=None,
+                 average=True, average_fr=False, oversampling=0,
+                 oversampling_fr=None, aligned=True, resample_filters_fr=True,
+                 out_type="array", out_3D=False, pad_mode='zero', max_pad_factor=2,
+                 max_pad_factor_fr=None, backend="torch"):
+        if oversampling_fr is None:
+            oversampling_fr = oversampling
+        # Second-order scattering object for the time variable
+        vectorize = True # for compatibility, will be removed in 0.3
+        max_order_tm = 2
+        scattering_out_type = out_type.lstrip('dict:')
+        ScatteringTorch1D.__init__(
+            self, J, shape, Q, T, max_order_tm, average, oversampling,
+            vectorize, scattering_out_type, pad_mode, max_pad_factor,
+            register_filters=False, backend=backend)
+
+        TimeFrequencyScatteringBase1D.__init__(
+            self, J_fr, Q_fr, F, average_fr, oversampling_fr, aligned,
+            resample_filters_fr, max_pad_factor_fr, out_3D, out_type)
+        TimeFrequencyScatteringBase1D.build(self)
+        self.register_filters()
+
+    def register_filters(self):
+        """ This function run the filterbank function that
+        will create the filters as numpy array, and then, it
+        saves those arrays as module's buffers."""
+        n_final = self._register_filters(self, ('phi_f', 'psi1_f'))
+        self._register_filters(self.sc_freq,
```

Comment:
what is `self.sc_freq` and where do I learn about it?
I would prefer these to be apparent in the code

Reply:
[here](https://github.com/kymatio/kymatio/blob/63e17b9e7a0183f4261ad3359dca8b8e6294fd44/kymatio/scattering1d/frontend/base_frontend.py#L817). Should I add an inline comment too?